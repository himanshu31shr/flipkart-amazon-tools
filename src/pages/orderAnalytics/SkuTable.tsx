import React from "react";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Collapse,
} from "@mui/material";

type SkuTable = {
  skus: { [sku: string]: { orders: number; returns: number, title: string } };
  open: boolean;
};

export const SkuTable: React.FC<SkuTable> = ({ skus, open }) => (
  <TableRow>
    <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={7}>
      <Collapse in={open} timeout="auto" unmountOnExit>
        <Table size="small">
          <TableHead>
            <TableRow>
              <TableCell>SKU</TableCell>
              <TableCell>Title</TableCell>
              <TableCell>Orders</TableCell>
              <TableCell>Returns</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {Object.entries(skus).map(([sku, counts]) => (
              <TableRow key={sku}>
                <TableCell>{sku}</TableCell>
                <TableCell>{counts.title}</TableCell>
                <TableCell>{counts.orders}</TableCell>
                <TableCell>{counts.returns}</TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </Collapse>
    </TableCell>
  </TableRow>
);
