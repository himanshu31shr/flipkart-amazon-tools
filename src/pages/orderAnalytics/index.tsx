import React, { useEffect, useState } from "react";
import {
  Box,
  Container,
  Grid,
  Paper,
  Typography,
  CircularProgress,
  IconButton,
  Badge,
  Button,
} from "@mui/material";
import FilterListIcon from "@mui/icons-material/FilterList";
import CalendarTodayIcon from "@mui/icons-material/CalendarToday";
import { format } from "date-fns";
import { useDispatch, useSelector } from "react-redux";
import { AppDispatch, RootState } from "../../store";
import { fetchAllOrdersForAnalytics } from "../../store/slices/allOrdersForAnalyticsSlice";
import { fetchProducts } from "../../store/slices/productsSlice";
import { fetchCategories } from "../../store/slices/categoriesSlice";
import { LocalizationProvider } from "@mui/x-date-pickers/LocalizationProvider";
import { AdapterDateFns } from "@mui/x-date-pickers/AdapterDateFns";
import CategoryDistributionTable from "./components/CategoryDistributionTable";
import CategoryOrdersChart from "./components/CategoryOrdersChart";
import OrderMetrics from "./components/OrderMetrics";
import SkuOrdersChart from "./components/SkuOrdersChart";
import TopProductsTable from "./components/TopProductsTable";
import DateRangeFilter from "./components/DateRangeFilter";
import FilterPopover from "./components/FilterPopover";
import { useOrderFilters } from "./hooks/useOrderFilters";

const OrderAnalytics: React.FC = () => {
  const dispatch = useDispatch<AppDispatch>();
  const {
    items: allOrders,
    loading: allOrdersLoading,
    error: allOrdersError,
  } = useSelector((state: RootState) => state.allOrdersForAnalytics);
  const {
    items: products,
    loading: productsLoading,
    error: productsError,
  } = useSelector((state: RootState) => state.products);
  const {
    items: categories,
    loading: categoriesLoading,
    error: categoriesError,
  } = useSelector((state: RootState) => state.categories);

  // State for popovers
  const [filterAnchorEl, setFilterAnchorEl] = useState<null | HTMLElement>(
    null
  );
  const [dateAnchorEl, setDateAnchorEl] = useState<null | HTMLElement>(null);

  // Use custom hook for filters
  const {
    filterState,
    skuOptions,
    platformOptions,
    productOptions,
    filteredOrders,
    updateFilter,
    clearFilters,
  } = useOrderFilters({ allOrders, categories, products });

  useEffect(() => {
    if (allOrders.length === 0 && !allOrdersLoading && !allOrdersError) {
      dispatch(fetchAllOrdersForAnalytics());
    }
    if (products.length === 0 && !productsLoading && !productsError) {
      dispatch(fetchProducts({}));
    }
    if (categories.length === 0 && !categoriesLoading && !categoriesError) {
      dispatch(fetchCategories());
    }
  }, []);

  const isLoading = allOrdersLoading || productsLoading || categoriesLoading;
  const error = allOrdersError || productsError || categoriesError;

  // Popover handlers
  const handleFilterClick = (event: React.MouseEvent<HTMLElement>) => {
    setFilterAnchorEl(event.currentTarget);
  };
  const handleFilterClose = () => {
    setFilterAnchorEl(null);
  };
  const handleDateClick = (event: React.MouseEvent<HTMLElement>) => {
    setDateAnchorEl(event.currentTarget);
  };
  const handleDateClose = () => {
    setDateAnchorEl(null);
  };

  // Active filter count
  const activeFilterCount = [
    filterState.selectedCategory,
    filterState.selectedSku,
    filterState.selectedPlatform,
    filterState.selectedProduct,
  ].filter(Boolean).length;

  // Date range summary
  const dateSummary = `${format(
    filterState.dateRange.startDate,
    "dd MMM yyyy"
  )} - ${format(filterState.dateRange.endDate, "dd MMM yyyy")}`;

  if (error) {
    return (
      <Container maxWidth="xl" sx={{ mt: 4, mb: 4 }}>
        <Typography color="error" variant="h6">
          Error: {error}
        </Typography>
      </Container>
    );
  }

  return (
    <LocalizationProvider dateAdapter={AdapterDateFns}>
      <Container maxWidth="xl" sx={{ mt: 4, mb: 4 }}>
        <Box
          display="flex"
          justifyContent="space-between"
          alignItems="center"
          mb={3}
        >
          <Typography variant="h4">Order Analytics</Typography>
          <Box display="flex" gap={2}>
            <Button
              variant="outlined"
              startIcon={<CalendarTodayIcon />}
              onClick={handleDateClick}
              sx={{ textTransform: "none" }}
            >
              {dateSummary}
            </Button>
            <IconButton
              onClick={handleFilterClick}
              color={activeFilterCount > 0 ? "primary" : "default"}
              sx={{ ml: 1 }}
            >
              <Badge badgeContent={activeFilterCount} color="primary">
                <FilterListIcon />
              </Badge>
            </IconButton>
          </Box>
        </Box>

        <DateRangeFilter
          dateRange={filterState.dateRange}
          onDateRangeChange={(startDate, endDate) =>
            updateFilter("dateRange", { startDate, endDate })
          }
          anchorEl={dateAnchorEl}
          onClose={handleDateClose}
        />

        <FilterPopover
          anchorEl={filterAnchorEl}
          onClose={handleFilterClose}
          categories={categories}
          selectedCategory={filterState.selectedCategory}
          selectedSku={filterState.selectedSku}
          selectedPlatform={filterState.selectedPlatform}
          selectedProduct={filterState.selectedProduct}
          skuOptions={skuOptions}
          platformOptions={platformOptions}
          productOptions={productOptions}
          onCategoryChange={(category) =>
            updateFilter("selectedCategory", category)
          }
          onSkuChange={(sku) => updateFilter("selectedSku", sku)}
          onPlatformChange={(platform) =>
            updateFilter("selectedPlatform", platform)
          }
          onProductChange={(product) =>
            updateFilter("selectedProduct", product)
          }
          onClearFilters={clearFilters}
        />

        {isLoading ? (
          <Box
            display="flex"
            justifyContent="center"
            alignItems="center"
            minHeight="400px"
          >
            <CircularProgress />
          </Box>
        ) : (
          <>
            <OrderMetrics
              orders={filteredOrders}
              products={products}
              categories={categories}
            />
            <Paper sx={{ p: 2 }}>
              <Grid container spacing={3}>
                <Grid item xs={12} md={7}>
                  <Typography variant="h6" gutterBottom>
                    Category Distribution Details
                  </Typography>
                  <CategoryDistributionTable
                    orders={filteredOrders}
                    categories={categories}
                  />
                </Grid>
                <Grid item xs={12} md={5}>
                  <Typography variant="h6" gutterBottom>
                    Orders by Category
                  </Typography>
                  <CategoryOrdersChart
                    orders={filteredOrders}
                    categories={categories}
                  />
                </Grid>
              </Grid>
            </Paper>
            <Paper sx={{ p: 2, mt:4, height: "100%" }}>
              <Grid container spacing={3}>
                <Grid item xs={12} md={7}>
                  <Typography variant="h6" gutterBottom>
                    Top Products
                  </Typography>
                  <TopProductsTable
                    orders={filteredOrders}
                    categories={categories}
                  />
                </Grid>
                <Grid item xs={12} md={5}>
                  <Typography variant="h6" gutterBottom>
                    Orders by SKU
                  </Typography>
                  <SkuOrdersChart
                    orders={filteredOrders}
                    categories={categories}
                  />
                </Grid>
              </Grid>
            </Paper>
          </>
        )}
      </Container>
    </LocalizationProvider>
  );
};

export default OrderAnalytics;

/* TODO: Create the CategoryDistributionChart component */
