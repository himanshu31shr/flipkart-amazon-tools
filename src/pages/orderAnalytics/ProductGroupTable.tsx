import React, { useState } from "react";
import {
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TableRow,
  Paper,
  IconButton,
  TableSortLabel,
  TableFooter,
  Typography,
} from "@mui/material";
import { KeyboardArrowDown, KeyboardArrowUp } from "@mui/icons-material";
import { SkuTable } from "./SkuTable";
import { Group } from "./types/group.type";

type ProductGroupTableProps = {
  groupedData: Group[];
  openRow: string | null;
  handleRowClick: (group: string) => void;
};

type Order = "asc" | "desc";

export const ProductGroupTable: React.FC<ProductGroupTableProps> = ({
  groupedData,
  openRow,
  handleRowClick,
}) => {
  const [order, setOrder] = useState<Order>("asc");
  const [orderBy, setOrderBy] = useState<keyof Group>("group");

  const handleRequestSort = (property: keyof Group) => {
    const isAsc = orderBy === property && order === "asc";
    setOrder(isAsc ? "desc" : "asc");
    setOrderBy(property);
  };

  const sortedData = [...groupedData].sort((a, b) => {
    if (orderBy === "group") {
      return order === "asc"
        ? a.group.localeCompare(b.group)
        : b.group.localeCompare(a.group);
    } else {
      return order === "asc"
        ? Number(a[orderBy]) - Number(b[orderBy])
        : Number(b[orderBy]) - Number(a[orderBy]);
    }
  });

  return (
    <TableContainer component={Paper}>
      <Table>
        <TableHead>
          <TableRow>
            <TableCell />
            <TableCell>
              <TableSortLabel
                active={orderBy === "group"}
                direction={orderBy === "group" ? order : "asc"}
                onClick={() => handleRequestSort("group")}
              >
                Group
              </TableSortLabel>
            </TableCell>
            <TableCell>
              <TableSortLabel
                active={orderBy === "orders"}
                direction={orderBy === "orders" ? order : "asc"}
                onClick={() => handleRequestSort("orders")}
              >
                Orders
              </TableSortLabel>
            </TableCell>
            <TableCell>
              <TableSortLabel
                active={orderBy === "returns"}
                direction={orderBy === "returns" ? order : "asc"}
                onClick={() => handleRequestSort("returns")}
              >
                Returns
              </TableSortLabel>
            </TableCell>
            <TableCell>
              <TableSortLabel
                active={orderBy === "returns"}
                direction={orderBy === "returns" ? order : "asc"}
                onClick={() => handleRequestSort("returns")}
              >
                Returns
              </TableSortLabel>
            </TableCell>
            <TableCell>
              <TableSortLabel
                active={orderBy === "returns"}
                direction={orderBy === "returns" ? order : "asc"}
                onClick={() => handleRequestSort("returns")}
              >
                Returns
              </TableSortLabel>
            </TableCell>
            <TableCell>
              <TableSortLabel
                active={orderBy === "returns"}
                direction={orderBy === "returns" ? order : "asc"}
                onClick={() => handleRequestSort("returns")}
              >
                Returns
              </TableSortLabel>
            </TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {sortedData.map((group) => (
            <React.Fragment key={group.group}>
              <TableRow onClick={() => handleRowClick(group.group)}>
                <TableCell>
                  <IconButton>
                    {openRow === group.group ? (
                      <KeyboardArrowUp />
                    ) : (
                      <KeyboardArrowDown />
                    )}
                  </IconButton>
                </TableCell>
                <TableCell>
                  <strong>{group.group.replaceAll("_", " ")}</strong>
                </TableCell>
                <TableCell>{group.orders}</TableCell>
                <TableCell>{group.returns}</TableCell>
                <TableCell>{group.returns}</TableCell>
                <TableCell>{group.returns}</TableCell>
                <TableCell>{group.returns}</TableCell>
              </TableRow>
              <SkuTable skus={group.skus} open={openRow === group.group} />
            </React.Fragment>
          ))}
        </TableBody>
        <TableFooter>
          <TableRow>
            <TableCell colSpan={2}>
              <Typography variant="h6" gutterBottom component="div">
                Totals
              </Typography>
            </TableCell>
            <TableCell>
              <Typography variant="h6" gutterBottom component="div">
                {sortedData.reduce((acc, curr) => acc + curr.orders, 0)}
              </Typography>
            </TableCell>
            <TableCell>
              <Typography variant="h6" gutterBottom component="div">
                {sortedData.reduce((acc, curr) => acc + curr.returns, 0)}
              </Typography>
            </TableCell>
          </TableRow>
        </TableFooter>
      </Table>
    </TableContainer>
  );
};
