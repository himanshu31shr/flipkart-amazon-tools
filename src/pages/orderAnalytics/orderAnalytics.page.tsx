import CloseIcon from "@mui/icons-material/Delete";
import { Alert, Box, CircularProgress, IconButton } from "@mui/material";
import Grid from "@mui/material/Grid2";
import Papa from "papaparse";
import React, { useEffect, useState } from "react";
import { deleteDatabase, getOrders } from "../../storage/db";
import { FileInput } from "../home/file-input";
import { ProductGroupTable } from "./ProductGroupTable";
import { OrderAggregationService } from "./services/orderAggregation.service";
import { Group } from "./types/group.type";
import { OrderItem } from "./types/order.type";

export const OrderAnalyticsPage = () => {
  const [orderItems, setOrderItems] = useState<OrderItem[]>([]);
  const [groupedData, setGroupedData] = useState<Group[]>([]);
  const [timeline, setTimeLine] = useState<
    Partial<{ start: string; end: string }>
  >({});

  const [fetchingDb, setFetchingDb] = useState(false);

  const fetchOrders = async () => {
    setFetchingDb(true);
    const savedOrders = await getOrders();
    if (savedOrders.length > 0) {
      setOrderItems(savedOrders);
      const service = new OrderAggregationService(savedOrders as OrderItem[]);
      setGroupedData(await service.process());
      setTimeLine(service.getDates());
    }
    setFetchingDb(false);
  };

  const deleteDb = () => {
    setFetchingDb(true);
    deleteDatabase();
    setOrderItems([]);
    setGroupedData([]);
    setTimeLine({});
    setFetchingDb(false);
  };

  useEffect(() => {
    fetchOrders();
  }, []);

  const handleFileUpload = (event: React.ChangeEvent<HTMLInputElement>) => {
    setFetchingDb(true);
    const file = event.target.files?.[0];
    if (file) {
      Papa.parse(file, {
        complete: async (results: Papa.ParseResult<OrderItem>) => {
          const data = results.data as OrderItem[];
          setOrderItems(data);
          const service = new OrderAggregationService(data);
          setGroupedData(await service.process());
          setTimeLine(service.getDates());
          setFetchingDb(false);
        },
        header: true,
        transform: (value: string, field: string) => {
          if (field === "sku") {
            value = value.replaceAll(/["]|sku:/gi, "");
          }

          return value;
        },
        skipEmptyLines: true,
      });
    }
  };

  return (
    <Loader showLoader={fetchingDb}>
      <Grid
        container
        spacing={2}
        justifyContent={"space-between"}
        marginTop={2}
        marginBottom={2}
      >
        <Box>
          <FileInput
            onChange={handleFileUpload}
            name="Flipkart Report"
            accepts=".csv"
            selected={orderItems == null}
          />
        </Box>
        <Box>
          <IconButton onClick={() => deleteDb()} data-testid="delete-data">
            <CloseIcon />{" "}
          </IconButton>
        </Box>
        <Grid size={12}>
          {timeline.start && timeline.end && (
            <Alert color="info">
              Showing data between {new Date(timeline.start).toDateString()} &{" "}
              {new Date(timeline.end).toDateString()}
            </Alert>
          )}
        </Grid>
      </Grid>
      {groupedData.length > 0 && (
        <ProductGroupTable
          groupedData={groupedData}
        />
      )}
    </Loader>
  );
};

const Loader = ({
  showLoader = false,
  children,
}: {
  showLoader: boolean;
  children: React.ReactNode;
}) => {
  return (
    <>
      {showLoader && (
        <Grid spacing={10} marginTop={2} alignItems={"center"}>
          <CircularProgress size={16} />{" "}Building data...
        </Grid>
      )}
      {children}
    </>
  );
};
