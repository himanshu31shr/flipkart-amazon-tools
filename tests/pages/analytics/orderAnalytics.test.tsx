import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";
import {
  getOrders,
  deleteDatabase,
  saveOrders,
  getOrderById,
} from "../../../src/storage/db";
import { OrderItem } from "../../../src/pages/orderAnalytics/types/order.type";
import { OrderAnalyticsPage } from "../../../src/pages/orderAnalytics/orderAnalytics.page";
import Papa from "papaparse";
import { MemoryRouter } from "react-router-dom";

// Mock the getOrders and deleteDatabase functions
jest.mock("../../../src/storage/db", () => ({
  getOrders: jest.fn(),
  deleteDatabase: jest.fn(),
  saveOrders: jest.fn(),
  getOrderById: jest.fn(),
}));

jest.mock("papaparse", () => ({
  parse: jest.fn(),
}));

const mockOrders: OrderItem[] = [
  {
    order_id: "1",
    sku: "SSPS001005",
    product_title: "Product 1",
    order_item_status: "Success",
    quantity: "1",
    order_date: "2023-01-01",
    order_item_id: "",
    fulfilment_source: "",
    fulfilment_type: "",
    order_approval_date: "",
    fsn: "",
    serial_no_imei: "",
    delivery_logistics_partner: "",
    pickup_logistics_partner: "",
    delivery_tracking_id: "",
    forward_logistics_form: "",
    forward_logistics_form_no: "",
    order_cancellation_date: "",
    cancellation_reason: "",
    cancellation_sub_reason: "",
    order_return_approval_date: "",
    return_id: "",
    return_reason: "",
    return_sub_reason: "",
    procurement_dispatch_sla: "",
    dispatch_after_date: "",
    dispatch_by_date: "",
    order_ready_for_dispatch_on_date: "",
    dispatched_date: "",
    dispatch_sla_breached: "",
    seller_pickup_reattempts: "",
    delivery_sla: "",
    deliver_by_date: "",
    order_delivery_date: "",
    delivery_sla_breached: "",
    order_service_completion_date: "",
    service_by_date: "",
    service_completion_sla: "",
    service_sla_breached: "",
  },
  {
    order_id: "2",
    sku: "SSPS002005",
    product_title: "Product 2",
    order_item_status: "Return",
    quantity: "2",
    order_date: "2023-01-02",
    order_item_id: "",
    fulfilment_source: "",
    fulfilment_type: "",
    order_approval_date: "",
    fsn: "",
    serial_no_imei: "",
    delivery_logistics_partner: "",
    pickup_logistics_partner: "",
    delivery_tracking_id: "",
    forward_logistics_form: "",
    forward_logistics_form_no: "",
    order_cancellation_date: "",
    cancellation_reason: "",
    cancellation_sub_reason: "",
    order_return_approval_date: "",
    return_id: "",
    return_reason: "",
    return_sub_reason: "",
    procurement_dispatch_sla: "",
    dispatch_after_date: "",
    dispatch_by_date: "",
    order_ready_for_dispatch_on_date: "",
    dispatched_date: "",
    dispatch_sla_breached: "",
    seller_pickup_reattempts: "",
    delivery_sla: "",
    deliver_by_date: "",
    order_delivery_date: "",
    delivery_sla_breached: "",
    order_service_completion_date: "",
    service_by_date: "",
    service_completion_sla: "",
    service_sla_breached: "",
  },
];

describe("OrderAnalyticsPage", () => {
  beforeEach(() => {
    (getOrders as jest.Mock).mockResolvedValue(mockOrders);
    (saveOrders as jest.Mock).mockResolvedValue(mockOrders);
    (getOrderById as jest.Mock).mockResolvedValue(mockOrders);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("renders without crashing", () => {
    render(<OrderAnalyticsPage />);
    expect(
      render(<OrderAnalyticsPage />).container
    ).toMatchSnapshot();
  });

  test("renders file input", () => {
    render(<OrderAnalyticsPage />);
    const fileInput = screen.getByLabelText(/flipkart report/i);
    expect(fileInput).toBeInTheDocument();
  });

  test("displays loader when fetching data", async () => {
    render(<OrderAnalyticsPage />);
    const loader = screen.getByRole("progressbar");
    expect(loader).toBeInTheDocument();
    await waitFor(() => expect(loader).not.toBeInTheDocument());
  });

  test("displays data after fetching", async () => {
    render(<OrderAnalyticsPage />);
    await waitFor(() => {
      const product1 = screen.getByText(/BHOG THALI/i);
      expect(product1).toBeInTheDocument();
    });
  });

  test("delete button works correctly", async () => {
    render(<OrderAnalyticsPage />);
    const deleteButton = screen.getByTestId("delete-data");
    fireEvent.click(deleteButton);
    await waitFor(() => {
      expect(deleteDatabase).toHaveBeenCalled();
      const product1 = screen.queryByText(/BHOG THALI/i);
      expect(product1).not.toBeInTheDocument();
    });
  });

  test("handleFileUpload processes file correctly", async () => {
    render(
      <MemoryRouter>
        <OrderAnalyticsPage />
      </MemoryRouter>
    );

    // Wait for data to load
    await screen.findByText(/Flipkart Report/);

    const file = new File(
      [
        `sku,orderDate,quantity,price
        sku1,2023-01-01,1,100
        sku2,2023-01-02,2,200`,
      ],
      "test.csv",
      { type: "text/csv" }
    );

    const input = screen.getByLabelText(/Flipkart Report/);
    fireEvent.change(input, { target: { files: [file] } });

    expect(Papa.parse).toHaveBeenCalled();
  });
});
