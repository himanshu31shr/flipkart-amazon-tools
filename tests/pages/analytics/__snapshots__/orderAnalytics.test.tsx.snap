// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderAnalyticsPage renders without crashing 1`] = `
<div>
  <div
    class="MuiGrid2-root MuiGrid2-direction-xs-row css-1ysf11g-MuiGrid2-root"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1wxecgq-MuiCircularProgress-root"
      role="progressbar"
      style="width: 16px; height: 16px;"
    >
      <svg
        class="MuiCircularProgress-svg css-54pwck-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-960umw-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
     
    Building data...
  </div>
  <div
    class="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-6q5qpn-MuiGrid2-root"
  >
    <div
      class="MuiBox-root css-0"
    >
      <label
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-colorPrimary css-1btmbba-MuiButtonBase-root-MuiButton-root"
        role="button"
        tabindex="-1"
      >
        <span
          class="MuiButton-icon MuiButton-startIcon MuiButton-iconSizeMedium css-1jsm74t-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
            data-testid="SquareOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 3v18h18V3zm16 16H5V5h14z"
            />
          </svg>
        </span>
        Flipkart Report
        <input
          accept=".csv"
          class="css-vwjrxt"
          type="file"
        />
      </label>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-aixiz9-MuiButtonBase-root-MuiIconButton-root"
        data-testid="delete-data"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
          data-testid="DeleteIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6zM19 4h-3.5l-1-1h-5l-1 1H5v2h14z"
          />
        </svg>
         
      </button>
    </div>
    <div
      class="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-1j0u08r-MuiGrid2-root"
    />
  </div>
</div>
`;
